zipkin-cassandra-core/src/main/java/org/twitter/zipkin/storage/cassandra/Repository.java

```diff
@@ -70,6 +70,23 @@
     private final PreparedStatement selectTraceDurationTail;
     private final Map<String,String> metadata;
 
+    private final ThreadLocal<Set<String>> writtenNames = new ThreadLocal<Set<String>>() {
+                private long hour = TimeUnit.MILLISECONDS.toHours(System.currentTimeMillis());
```

Comment:
a couple of problems with this cache. First, why one hour? Different installations might use different TTL in Cassandra, in most cases at least 1 day, so cache TTL of 1hr is too fine. I have similar logic in Go collector, we use 12hr TTL for cache. In this OSS implementation I'd suggest making this TTL externally configurable.

Second problem - the TTL is on the whole cache, not on individual entries. It means once an hour *all threads* in *all collectors* will drop the cache and re-insert unnecessary data. 

Granted, both are not huge issues given the ratio of writes, but the implementation would not be much more difficult if both were addressed.

Also, thread locals are unnecessary here, a concurrent map shared between threads would do.